integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py

```diff
@@ -0,0 +1,199 @@
+                aws_session_token=os.getenv(self.aws_session_token_var),
+                region_name=os.getenv(self.aws_region_name_var),
+                profile_name=os.getenv(self.aws_profile_name_var),
+            )
+            self.client = session.client("sagemaker-runtime")
+        except Exception as e:
+            msg = (
+                f"Could not connect to SageMaker Inference Endpoint '{self.model}'."
+                f"Make sure the Endpoint exists and AWS environment is configured."
+            )
+            raise AWSConfigurationError(msg) from e
+
+    @component.output_types(replies=List[str], meta=List[Dict[str, Any]])
+    def run(self, prompt: str, generation_kwargs: Optional[Dict[str, Any]] = None):
+        """
+        Invoke the text generation inference based on the provided messages and generation parameters.
+
+        :param prompt: The string prompt to use for text generation.
+        :param generation_kwargs: Additional keyword arguments for text generation. These parameters will
+        potentially override the parameters passed in the `__init__` method.
+
+        :return: A list of strings containing the generated responses and a list of dictionaries containing the metadata
+        for each response.
+        """
+        if self.client is None:
+            msg = "SageMaker Inference client is not initialized. Please call warm_up() first."
+            raise ValueError(msg)
+
+        generation_kwargs = generation_kwargs or self.generation_kwargs
+        custom_attributes = ";".join(
+            f"{k}={str(v).lower() if isinstance(v, bool) else str(v)}" for k, v in self.aws_custom_attributes.items()
+        )
+        try:
+            body = json.dumps({"inputs": prompt, "parameters": generation_kwargs})
+            response = self.client.invoke_endpoint(
+                EndpointName=self.model,
+                Body=body,
+                ContentType="application/json",
+                Accept="application/json",
+                CustomAttributes=custom_attributes,
+            )
+            response_json = response.get("Body").read().decode("utf-8")
+            output: Dict[str, Dict[str, Any]] = json.loads(response_json)
+
+            # The output might be either a list of dictionaries or a single dictionary
+            list_output: List[Dict[str, Any]]
+            if output and isinstance(output, dict):
+                list_output = [output]
+            elif isinstance(output, list) and all(isinstance(o, dict) for o in output):
+                list_output = output
+            else:
+                msg = f"Unexpected model response type: {type(output)}"
+                raise ValueError(msg)
+
+            # The key where the replies are stored changes from model to model, so we need to look for it.
+            # All other keys in the response are added to the metadata.
+            for key in self.model_generation_keys:
```

Comment:
Yeah, I think we can forget about matching the meta format here that we've strived to match so far. But hey, perhaps leave a code comment about that here...